src/Database/LSMTree/Internal/BlobRef.hs

```diff
@@ -89,82 +116,98 @@ newtype WeakBlobRefInvalid = WeakBlobRefInvalid Int
 withWeakBlobRef ::
      (MonadMask m, PrimMonad m)
   => WeakBlobRef m h
-  -> (RawBlobRef m h -> m a)
+  -> (StrongBlobRef m h -> m a)
   -> m a
 withWeakBlobRef wref = bracket (deRefWeakBlobRef wref) removeReference
 
 {-# SPECIALISE withWeakBlobRefs ::
      V.Vector (WeakBlobRef IO h)
-  -> (V.Vector (RawBlobRef IO h) -> IO a)
+  -> (V.Vector (StrongBlobRef IO h) -> IO a)
   -> IO a #-}
 -- | The same as 'withWeakBlobRef' but for many references in one go.
 --
 withWeakBlobRefs ::
      (MonadMask m, PrimMonad m)
   => V.Vector (WeakBlobRef m h)
-  -> (V.Vector (RawBlobRef m h) -> m a)
+  -> (V.Vector (StrongBlobRef m h) -> m a)
   -> m a
 withWeakBlobRefs wrefs = bracket (deRefWeakBlobRefs wrefs) removeReferences
 
 {-# SPECIALISE deRefWeakBlobRef ::
      WeakBlobRef IO h
-  -> IO (RawBlobRef IO h) #-}
+  -> IO (StrongBlobRef IO h) #-}
 deRefWeakBlobRef ::
      (MonadThrow m, PrimMonad m)
   => WeakBlobRef m h
-  -> m (RawBlobRef m h)
-deRefWeakBlobRef (WeakBlobRef ref) = do
-    ok <- RC.upgradeWeakReference (blobFileRefCounter (rawBlobRefFile ref))
+  -> m (StrongBlobRef m h)
+deRefWeakBlobRef WeakBlobRef{weakBlobRefFile, weakBlobRefSpan} = do
+    ok <- RC.upgradeWeakReference (blobFileRefCounter weakBlobRefFile)
     when (not ok) $ throwIO (WeakBlobRefInvalid 0)
-    pure ref
+    return StrongBlobRef{
+      strongBlobRefFile = weakBlobRefFile,
+      strongBlobRefSpan = weakBlobRefSpan
+    }
```

Comment:
Maybe we should rename this to `upgradeWeakBlobRef` instead, following the naming from the `Control.RefCount` module?

Reply:
The `Ref` API will have a ```deRefWeak :: WeakRef a -> m (Maybe (Ref a))```